vllm/distributed/device_communicators/pynccl.py

```diff
@@ -180,8 +179,8 @@ def from_torch(cls, op: ReduceOp) -> 'ncclRedOp_t':
 _c_ncclAllReduce = nccl.ncclAllReduce
 _c_ncclAllReduce.restype = ctypes.c_int
 _c_ncclAllReduce.argtypes = [
-    ctypes.c_void_p, ctypes.c_void_p, ctypes.c_size_t, ncclDataType_t,
-    ncclRedOp_t, ctypes.c_void_p, ctypes.c_void_p
+    ctypes.c_void_p, ctypes.c_void_p, ctypes.c_size_t, ctypes.c_int,
+    ctypes.c_void_p, ctypes.c_void_p
```

Comment:
This is not intuitive. Let's keep `ncclDataType_t` and `ncclRedOp_t` here.

Does mypy understand the call signature of `_c_ncclAllReduce`? I suppose it will just ignore them. And we can leave a comment here, saying that `int` will be automatically converted by `ctypes`.

Reply:
I found ncclDataType_t being a random enum and type at the same time a bit weird though. Why don't we then just

```
ncclDataType_t(ctype.c_int):
    pass

ncclDataEnum:
    ... = 0
    .... = 1
```
?

Reply:
Well, that's also doable.

Please use `ncclDataType_t  = ctype.c_int`, and `ncclDataTypeEnum`

Reply:
sgtm!